tensorflow/core/util/gpu_kernel_helper.h

```diff
@@ -359,13 +358,12 @@ __device__ OutType upper_bound(const T* first, OutType count, T val) {
   return first - orig;
 }
 
-template <typename T, typename OutType = int32>
-__device__ OutType lower_bound(const T* first, OutType count, T val) {
-  const T* orig = first;
-  const T* it = nullptr;
+template <typename T, typename OutType = int32, typename Iterator = const T*>
+__device__ OutType lower_bound(Iterator first, OutType count, T val) {
```

Comment:
Add a comment, since it's a bit unclear what this does. E.g., "Returns the first element in [first, first+count) that is not less than "val". Assumes [first, first+count) is sorted."

Also similarly for upper_bound.

Reply:
Done.